knimin/static/js/platemapper.js

```diff
@@ -269,36 +303,137 @@ PlateMap.prototype.keypressWell = function (current, e) {
 
 /**
  *
+ * Updates the GUI to show the technical replicates
+ *
+ * @param {Object} sampleInfo The sample information object
+ *
+ **/
+PlateMap.prototype._updateTechnicalReplicate = function (sampleInfo) {
+  if (sampleInfo.wells.length > 1) {
+    // Update the list of technical replicates
+    if ($.inArray(sampleInfo, this.technicalReplicates) == -1) {
+      this.technicalReplicates.push(sampleInfo);
+    }
+    // Mark all wells as technical
+    for (var well of sampleInfo.wells) {
+      this.wellInformation[well[0]][well[1]].inputTag.addClass('pm-technical-replicate');
+    }
+  } else {
+    // Using a for loop to avoid to special case - If the wells list is
+    // empty, this for loop will not execute anything. If there is one element
+    // it will correctly remove the mark for techincal replicates
+    for (var well of sampleInfo.wells) {
+      this.wellInformation[well[0]][well[1]].inputTag.removeClass('pm-technical-replicate');
+    }
+    // Remove from the techincal replicates list
+    this.technicalReplicates = $.grep(this.technicalReplicates, function(value) {
+      return value != sampleInfo;
+    });
+  }
+}
+
+/**
+ *
+ * Updates the GUI to show the error list
+ *
+ **/
+PlateMap.prototype.updateErrorList = function () {
+  var well, message, $li;
+  // Empty the current list
+  $("#pm-error-list").html("");
+
+  // First look for any warning/error in a per well basis
+  for (var i = 0; i < this.wellInformation.length; i++) {
+    for (var j = 0; j < this.wellInformation[i].length; j++) {
+      well = this.wellInformation[i][j];
+      if (well.inputTag.hasClass('pm-wrong-sample')) {
+        // Add an error to the list
+        message = 'Well ' + this._formatWellId(i, j) + ': Sample "' + well.inputTag.val().trim() + '" not found in any study';
+        $('<li>').addClass('list-group-item').addClass('list-group-item-danger').html(message).appendTo('#pm-error-list');
+      } else if (well.inputTag.hasClass('pm-sample-plated')) {
+        // Add a warning to the list
+        sample = well.inputTag.val().trim();
+        message = 'Well ' + this._formatWellId(i, j) + ': Sample "' + sample + '" previously plated in plate(s): ';
+        $li = $('<li>').addClass('list-group-item').addClass('list-group-item-warning').html(message).appendTo('#pm-error-list');
+        // Add a link for each plate that the sample is already plated
+        for (var plate of this.samples[sample].plates) {
+          $('<a>').attr('href', '/pm_plate_map?plate_id=' + plate).attr('target', '_blank').text(plate + ' ').appendTo($li);
+        }
+      }
+    }
+  }
+
+  for (var sampleInfo of this.technicalReplicates) {
```

Comment:
I think this `for ... of` thing is a new feature of the recent ECMAScript 6 standard. May have compatibility issues with old computers?